### ISSUE TITLE
SMG turret wastes ammo at range

### ISSUE THREAD

**Issue by Coolthulhu** [CleverRaven/Cataclysm-DDA#16012]
Could be a result of ammo nerf or a bug in gun actor. SMG turret doesn't have enough damage to shoot at the range it tries to shoot. The projectiles will drop to 0 damage before reaching the target.

The turret on top of the lab is a good example: it doesn't have enough power to reach player on top of the stairs, but will waste all ammo trying to do so.

EDIT: Could also be caused by too heavy damage drop due to range. Rounding errors?

EDIT 2: It's actually because gun actor has range 18, while the gun+ammo combo has range 14. Ranges are now enforced (they weren't before).

**Comment by player0981234** [comment]
just reduce the detection range of the smg turret

**Comment by Labtop-215** [comment]
Why does the damage from fired ammunition drop beyond max range?  Is the bullet slowing down in the air _Matrix Style_?

I would think that firing at a target at max range should simply make the target harder to hit simply because that's just more distance that if the bullet veers from, would miss.  Simply put, this is something that should be handled by dispersion.

**Comment by Coolthulhu** [comment]
Damage drops with distance because air resistance.

**Comment by mugling** [comment]
> Damage drops with distance because air resistance.

Whilst that's entirely true in our current implementation `range` is entirely irrelevant due to the effects of `damage` and `dispersion`. I'm working on some changes to this at the moment

**Comment by Labtop-215** [comment]
Does it affect every type of bullet fired the same way?  Are there some bullets that are more aerodynamic than others?  Also, shouldn't there be some sort of cap on how badly your damage has been reduced then?  Currently, it seems like every tile the bullet travels through is some sort of solid paper wall instead of just air.

**Comment by Rivet-the-Zombie** [comment]
I think y'all are overestimating how rapidly wind resistance can slow a bullet in flight. There really shouldn't be any appreciable effect when fired at close targets, like is happening in this case.

**Comment by Coolthulhu** [comment]
Those ranges aren't really all that close - this is near the maximum possible range of a 9mm SMG.

Wait, it's actually beyond the range of a 9mm SMG. That's why it disappears.

**Comment by Karbine** [comment]
I think Rivet was referring to range as it pertains to ballistic performance in meatspace. IRL there's not much appreciable difference between the kinetic energy of a 9mm striking a target at 1m or 30m.

If damage fall-off is so severe that it drops to zero within those ranges, maybe that's something to look at with the way projectile damage is modeled.

**Comment by Coolthulhu** [comment]
> IRL there's not much appreciable difference between the kinetic energy of a 9mm striking a target at 1m or 30m.

But we're talking about maximum feasible range of an SMG here. Gun ranges aren't 1-1 with real ones, because otherwise most guns would have ranges larger than the reality bubble.

**Comment by Labtop-215** [comment]
> But we're talking about maximum feasible range of an SMG here. Gun ranges aren't 1-1 with real ones, because otherwise most guns would have ranges larger than the reality bubble.

Maybe they should.  All you really need is an indicator that you are aiming at a target too far away to have a good chance to hit that target, and to start applying additional penalties, and maybe a chance for the bullet to actually hit the ground short of the target in extreme cases, or go wildly beyond the target.

**Comment by Coolthulhu** [comment]
> Maybe they should.

Maybe in a mod.

It doesn't work for gameplay. It's one of those cases where realism vs. gameplay is pretty much a zero sum game.

**Comment by Labtop-215** [comment]
> It doesn't work for gameplay. It's one of those cases where realism vs. gameplay is pretty much a zero sum game.

How does allowing for realistic ranges hurt gameplay?  Guns in general are meant to be fired at targets that are a long way off.  Some guns are better at close range like handguns that are meant mainly for personal defense, and some guns are much better at long range combat, like hunting rifles.  But I don't see why you couldn't try to shoot at a target that is a long way off.  The worst that should happen is that you miss, or that your gun misfires (which would have happened anyway if it was going too).

The only adjustments we would really have to make are the addition of an indicator that your shooting at something outside of the weapons normal range, and for really long range combat, we'll need an indicator of where an NPC is threatening you from if they are threatening you at long range with a sniper rifle.  (Assuming you spot them.  I don't know why you wouldn't.)  Also I'm not talking about combat coming in from outside the reality bubble, since that would take being able to simulate stuff outside of the reality bubble.

**Comment by Coolthulhu** [comment]
> How does allowing for realistic ranges hurt gameplay?

It would make:
* Sniper rifles useless, because their primary role could be fulfilled by short-ranged pistols just fine
* Shotguns the masters of all trades, capable of taking down everything from everywhere
* Pistols, which are already very strong, the second best choice at almost everything (and the best choice at some things)
* SMGs even more useless than they are now, by not letting anyone approach you, meaning you'd never really need to use burst fire
* Ranges of pistols and sniper rifles practically identical, because range 30 (still not "realistic" for a pistol) is more than enough for all purposes except outranging turrets and very long range sniping

Basically, making ranges realistic means removing them from the equation once you mod and accurize your stuff enough. And for some weapons (shotguns) that's very early.

**Comment by Labtop-215** [comment]
That's what dispersion penalties are for outside of ideal range.  Weapons meant for long range combat shouldn't receive any additional penalties for long ranged combat, while pistols would have a hard time hitting a target at long range.  A pistol could still be used at long range, if ammunition waste wasn't an issue (which it generally is).  I also don't see why burst fire from an assault rifle or an uzi wouldn't be at least useful if you where desperate.  Generally burst fire is a bad idea anyway though because of how wasteful it is.

If your using a shotgun with ammo that is meant to spread when fired, the current system would make sense in this _particular_ case.

Also, why would it not be a two way street?  Why wouldn't a turret be able to shoot you at the same range you are shooting it from?  Since it's a computer essentially, shouldn't it be substantially better than the average person at hitting their mark as well?  Are they subject to dispersion penalties like people are?  Maybe they shouldn't have so many penalties because that's a machine that both, isn't moving, and is equipped with at least at basic (and we're generally talking military grade, unless the survivor has crafted them) A.I. that is dedicated to finding and shooting at live targets.

Also, does re-enforcing your weapons still buff their attributes?  I know that was removed from armor.

**Comment by illi-kun** [comment]
I'm with Coolthulhu here. Current system is balanced with assumption that using of weapon for targets beyond the effective range is pure wasting of ammo. Shooting to such target is not allowed for convinience. Single issue with turret should not cause total rebalance, it should be simply fixed to match the rest of weaponry.

**Comment by kevingranade** [comment]
This isn't the place to discuss overhauling the entire ranged combat system, probably the simplest fix is to reduce the targeting range of the turret AI, but if that makes them too easy to snipe then it needs a gun with a better range.
